broker/cmd/base/main.go

```diff
@@ -145,9 +147,34 @@ func loadConfig() (Config, error) {
 	}
 
 	err = v.Unmarshal(&cfg)
+	overrideConfigByCLIParams(&cfg)
+
 	return cfg, err
 }
 
+func overrideConfigByCLIParams(cfg *Config) {
+	s := reflect.ValueOf(cfg).Elem()
+	for i := 0; i < s.NumField(); i++ {
+		tagName, ok := s.Type().Field(i).Tag.Lookup("mapstructure")
+		if ok && s.Field(i).CanSet() {
```

Comment:
please refactor this by extracting the second for in a separate func and also update the IFs to decrease the number indentation levels